vlog4j-graal/src/test/java/org/semanticweb/vlog4j/graal/GraalToVLog4JModelConverterTest.java

```diff
@@ -0,0 +1,106 @@
+package org.semanticweb.vlog4j.graal;
+
+/*-
+ * #%L
+ * VLog4J Graal Import Components
+ * %%
+ * Copyright (C) 2018 VLog4j Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import static org.junit.Assert.assertEquals;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeAtom;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConjunction;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConstant;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makePredicate;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeRule;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeVariable;
+
+import org.junit.Test;
+import org.semanticweb.vlog4j.core.model.api.Atom;
+import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Variable;
+
+import fr.lirmm.graphik.graal.api.io.ParseException;
+import fr.lirmm.graphik.graal.io.dlp.DlgpParser;
+
+/**
+ * @author Adrian Bielefeldt
+ */
+public class GraalToVLog4JModelConverterTest {
+	
+	private final String bicycle = "bicycle";
+	private final String hasPart = "hasPart";
+	private final String human = "human";
+	private final String mortal = "mortal";
+	private final String socrate = "socrate";
+	private final String wheel = "wheel";
+	
+	private final String x = "X";
+	private final String y = "Y";
+	
+	private final Constant vlog4j_socrate = makeConstant(socrate);
+	
+	private final Predicate vlog4j_bicycle = makePredicate(bicycle, 1);
+	private final Predicate vlog4j_hasPart = makePredicate(hasPart, 2);
+	private final Predicate vlog4j_mortal = makePredicate(mortal, 1);
+	private final Predicate vlog4j_human = makePredicate(human, 1);
+	private final Predicate vlog4j_wheel = makePredicate(wheel, 1);
+	
+	private final Variable vlog4j_x = makeVariable(x);
+	private final Variable vlog4j_y = makeVariable(y);
+
```

Comment:
even though Dlgp parser does not allow atoms with empty bodies or rules with empty heads, it should be tested that the conversion throws an exception in this case. graal-core (within test scope) DefaultAtom, DefaultRule, DefaultConjunctiveQuery can be used for testing, in this case.

Reply:
It would actually be more straight-forward and readable to use Predicate, DefaultAtom, DefaultRule and DefaultConjunctiveQuery to create graal objects for testing, instead of of obtaining them by parsing.

Reply:
Rewrote the tests to use constructors and factory methods instead of parsing.